sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -69,13 +69,19 @@ private[hive] class HiveMetastoreCatalog(hive: HiveContext) extends Catalog with
         val table = synchronized {
           client.getTable(in.database, in.name)
         }
-        val schemaString = table.getProperty("spark.sql.sources.schema")
+        val schemaString = Option(table.getProperty("spark.sql.sources.schema.numOfParts")) match {
```

Comment:
I think it is more conventional to use numParts instead of numOfParts. Also you can remove the pattern matching by just applying a map.
 
```scala
Option(table.getProperty("spark.sql.sources.schema.numParts")).map { numParts =>
  ...
}
```